src/execution_plan/ops/shared/update_functions.c

```diff
@@ -272,6 +272,28 @@ void EvalEntityUpdates(GraphContext *gc, PendingUpdateCtx **node_updates,
 				array_append(*updates, update);
 			}
 			continue;
+		} else if(SI_TYPE(new_value) & (T_NODE | T_EDGE)) {
```

Comment:
where's the distinction between `a=b` and `a+=b` ?

Reply:
This is managed earlier, by (in the first case) enqueuing a dummy update that clears the existing attribute set - https://github.com/RedisGraph/RedisGraph/blob/master/src/execution_plan/ops/shared/update_functions.c#L228-L237